src/OrchardCore/OrchardCore.Deployment.Abstractions/ServiceCollectionExtensions.cs

```diff
@@ -1,16 +1,59 @@
 using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.DisplayManagement.Handlers;
 
-namespace OrchardCore.Deployment
+namespace OrchardCore.Deployment;
+
+public static class ServiceCollectionExtensions
 {
-    public static class ServiceCollectionExtensions
+    public static IServiceCollection AddDeploymentTargetHandler<TImplementation>(this IServiceCollection serviceCollection)
+        where TImplementation : class, IDeploymentTargetHandler
+    {
+        serviceCollection.AddScoped<IDeploymentTargetHandler, TImplementation>();
+
+        return serviceCollection;
+    }
+
+    public static void AddDeployment<TSource, TStep>(this IServiceCollection services)
```

Comment:
`public static IServiceCollection AddDeployment<TSource, TStep>(this IServiceCollection services)`

allow chaining

Reply:
@deanmarcussen added